### PR TITLE
feat: add Android 19 compatible FileDataStoreFactory implementation

### LINKED ISSUE #906: google-http-client-android 1.31+ Posix crash
> com.google.api.client.util.store.FileDataStoreFactory.setPermissionsToOwnerOnly

uses `PosixFilePermission`
This class was added only in Android API 26. Running on lower API devices results in:

> Fatal Exception: java.lang.NoClassDefFoundError
Failed resolution of: Ljava/nio/file/attribute/PosixFilePermission;
com.google.api.client.util.store.FileDataStoreFactory.setPermissionsToOwnerOnly

Crash happens when trying to store OAuth credentials and doing something trivial like

> new FileDataStoreFactory(filesDir)

Maybe since the package releases as "Android-compatible" this should be documented?

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #906

* Adds an Android compatible FileDataStoreFactory implementation (version prior to switching to NIO) in the google-http-client-android extensions artifact
* Adds a note in the core FileDataStoreFactory to use the new version on Android API level < 26
* Adds an animal-sniffer check for Android API level 19 (Ignores the NIO Files API usage).

### CODE REVIEW

File: google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    } catch (InvocationTargetException exception) {
+      Throwable cause = exception.getCause();
+      Throwables.propagateIfPossible(cause, IOException.class);
+      // shouldn't reach this point, but just in case...
+      throw new RuntimeException(cause);
+    } catch (NoSuchMethodException exception) {
+      LOGGER.warning(
+          "Unable to set permissions for "
+              + file
+              + ", likely because you are running a version of Java prior to 1.6");
```
Comment (elharo) [inline google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java:152]: Is this the right error message here?

Comment (chingor13) [reply google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java:152]: Probably not, this was the version from google-http-client (not restricted to android). I'll see if I can come up with a better message.

Comment (chingor13) [reply google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java]: Removed the reflection as the methods we were previously protecting were added in [API level 9](https://developer.android.com/reference/java/io/File#setReadable(boolean,%2520boolean)).

File: google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.logging.Logger;
+
+/**
+ * Thread-safe file implementation of a credential store.
+ *
+ * <p>For security purposes, the file's permissions are set to be accessible only by the file's
+ * owner. Note that Java 1.5 does not support manipulating file permissions, and must be done
```
Comment (elharo) [inline google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java:34]: no need to mention Java 1.5 any more

File: google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+  /** Directory to store data. */
+  private final File dataDirectory;
+
+  /** @param dataDirectory data directory */
+  public FileDataStoreFactory(File dataDirectory) throws IOException {
+    dataDirectory = dataDirectory.getCanonicalFile();
+    this.dataDirectory = dataDirectory;
+    // error if it is a symbolic link
+    if (IOUtils.isSymbolicLink(dataDirectory)) {
```
Comment (elharo) [inline google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java:55]: the object has been modified before the exception is thrown

File: google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java (hunk 2 above)
Comment (chingor13) [reply google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java]: Done, removed.

File: google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java (hunk 3 above)
Comment (chingor13) [reply google-http-client-android/src/main/java/com/google/api/client/extensions/android/util/store/FileDataStoreFactory.java:53]: Done. Moved the member assignment after the validations here and in the mainline implementation.

**Pull request merged**